File: frontend/src/graphql/todo.ts
import { gql } from '@apollo/client';

export const GET_TODOS = gql`
  query GetTodos {
    todos {
      id
      title
      description
      completed
      createdAt
    }
  }
`;

export const GET_TODO = gql`
  query GetTodo($id: Int!) {
    todo(id: $id) {
      id
      title
      description
      completed
      createdAt
    }
  }
`;

export const CREATE_TODO = gql`
  mutation CreateTodo($title: String!, $description: String!) {
    createTodo(title: $title, description: $description) {
      todo {
        id
        title
        description
        completed
      }
      success
      message
    }
  }
`;

export const UPDATE_TODO = gql`
  mutation UpdateTodo(
    $id: Int!
    $title: String
    $description: String
    $completed: Boolean
  ) {
    updateTodo(
      id: $id
      title: $title
      description: $description
      completed: $completed
    ) {
      todo {
        id
        title
        description
        completed
      }
      success
      message
    }
  }
`;

export const DELETE_TODO = gql`
  mutation DeleteTodo($id: Int!) {
    deleteTodo(id: $id) {
      success
      message
    }
  }
`;

File: backend/src/domain/models/todo.py
from datetime import datetime
from typing import Optional

class Todo:
    def __init__(
        self,
        id: Optional[int],
        title: str,
        description: str,
        completed: bool,
        user_id: int,
        created_at: Optional[datetime] = None
    ):
        self.id = id
        self.title = title
        self.description = description
        self.completed = completed
        self.user_id = user_id
        self.created_at = created_at or datetime.utcnow()
